src/puppetlabs/puppetdb/query/summary_stats.clj

```diff
@@ -0,0 +1,187 @@
+    within group (order by length) quantiles 
+    from
+    (select char_length(metrics::text) as length from reports
+    where metrics is not null) foo"
+
+   :report_log_size_dist
+   "select percentile_cont(Array[0, 0.05, 0.10, 0.15, 0.20, 0.25, 0.30,
+    0.35, 0.40, 0.45, 0.50, 0.55, 0.60,
+    0.65, 0.70, 0.75, 0.80, 0.85, 0.90, 0.95, 1])
+    within group (order by length) quantiles 
+    from
+    (select char_length(logs::text) as length from reports
+    where logs is not null) foo" 
+
+   :fact_values_by_type
+   "select vt.type, count(*)
+    from fact_values fv inner join value_types vt on fv.value_type_id = vt.id
+    group by vt.type" 
+   
+   :num_associated_factsets_over_fact_paths
+   "select percentile_cont(Array[0, 0.05, 0.10, 0.15, 0.20, 0.25, 0.30,
+    0.35, 0.40, 0.45, 0.50, 0.55, 0.60,
+    0.65, 0.70, 0.75, 0.80, 0.85, 0.90, 0.95, 1])
+    within group (order by c) quantiles 
+    from
+    (select fact_path_id, count(*) as c
+    from facts
+    group by fact_path_id
+    order by c) foo" 
+   
+   :num_resources_per_node
+   "select percentile_cont(Array[0, 0.05, 0.10, 0.15, 0.20, 0.25, 0.30,
+    0.35, 0.40, 0.45, 0.50, 0.55, 0.60,
+    0.65, 0.70, 0.75, 0.80, 0.85, 0.90, 0.95, 1])
+    within group (order by c) quantiles 
+    from
+    (select count(*) as c from catalog_resources group by certname_id) foo"
+   
+   :num_resources_per_file
+   "select percentile_cont(Array[0, 0.05, 0.10, 0.15, 0.20, 0.25, 0.30,
+    0.35, 0.40, 0.45, 0.50, 0.55, 0.60,
+    0.65, 0.70, 0.75, 0.80, 0.85, 0.90, 0.95, 1])
+    within group (order by c) quantiles 
+    from
+    (select count(*) as c from catalog_resources where file is not null group by file) foo"
+
+   :file_resources_per_catalog
+   "select percentile_cont(Array[0, 0.05, 0.10, 0.15, 0.20, 0.25, 0.30,
+    0.35, 0.40, 0.45, 0.50, 0.55, 0.60,
+    0.65, 0.70, 0.75, 0.80, 0.85, 0.90, 0.95, 1])
+    within group (order by c) quantiles
+    from
+    (select count(*) as c, certname_id from catalog_resources where type='File'
+    group by certname_id) foo"
+
+   :file_resources_per_catalog_with_source
+   "select percentile_cont(Array[0, 0.05, 0.10, 0.15, 0.20, 0.25, 0.30,
+    0.35, 0.40, 0.45, 0.50, 0.55, 0.60,
+    0.65, 0.70, 0.75, 0.80, 0.85, 0.90, 0.95, 1])
+    within group (order by c) quantiles
+    from
+    (select count(*) as c, certname_id from catalog_resources
+    inner join resource_params on
+    catalog_resources.resource=resource_params.resource
+    where type='File' and name='source'
+    group by certname_id) foo"
+   
+   :num_distinct_edges_source_target
+   "select count(distinct (source, target)) from edges"})
+
+(defn collect-metadata
+  [get-shared-globals]
+  (let [{:keys [scf-read-db] :as db} (get-shared-globals)]
+    (jdbc/with-transacted-connection scf-read-db
+      (->> (keys metadata-queries)
+           (reduce #(assoc %1 %2 (jdbc/query-to-vec (%2 metadata-queries)))
+                   {:version (v/version)})
```

Comment:
`(assoc (kitchensink/mapvals jdbc/query-to-vec metadata-queries) :version (v/version))`